src/components/Inputs/TaxIdInput/TaxIdInput.vue

```diff
@@ -0,0 +1,230 @@
+<template>
+  <div class="pb-document-input-container">
+    <input
+      v-model="taxIdInput"
+      class="pb"
+      style="width: 100%"
+      :class="validationClass"
+      :style="taxIdInputStyle"
+      :placeholder="setInputPlaceholder"
+      :maxlength="getMaxLength(inputType)"
+      @blur="validateDocument"
+      @focus="updateValidationFiled"
+    >
+    <p
+      v-if="state.message"
+      class="pb error-text"
+    >
+      {{ state.message }}
+    </p>
+  </div>
+</template>
+
+<script>
+import { validateColor } from '@pb/utils/validator';
+import { documentPlaceholders } from '@pb/utils/inputDocumentTypes';
+import { isCpf, isCnpj } from 'adapcon-utils-js';
+
+export default {
+  name: 'PbTaxIdInput',
+
+  props: {
+    required: { type: Boolean, default: false },
+    value: { type: String, default: '' },
+    background: { type: String, default: 'transparent' },
+    color: { type: String, default: 'gray-20', validator: validateColor },
+    inputType: {
+      type: Array,
+      default: () => ['cnpj'],
+    },
+    placeholder: { type: String, default: '__.___.___/____-__' },
+  },
+
+  data() {
+    return {
+      state: {
+        validation: true,
+        message: '',
+        inputValue: '',
+      },
+    };
+  },
+
+  computed: {
+    validationClass() {
+      return !this.state.validation ? 'error' : '';
+    },
+    taxIdInputStyle() {
+      return {
+        ...(this.state.validation
+          ? {
+            background: `var(--color-${this.background})`,
+            border: `1px solid var(--color-${this.color})`,
+          }
+          : { background: `var(--color-${this.background})` }),
+      };
+    },
+
+    setInputPlaceholder() {
+      if (!this.placeholder && this.inputType.length <= 1)
+        return documentPlaceholders[this.inputType[0]];
+
+      return this.placeholder;
+    },
+
+    setLengthValidator() {
+      const documentLength = [];
+
+      if (this.inputType.includes('cnpj')) documentLength.push('14');
+
+      if (this.inputType.includes('cpf')) documentLength.push('11');
+
+      const lengthValidatorGenerate = this.inputType.map((doc, index) => {
+        if (index === this.inputType.length - 1)
+          return `documentToValidate.length !== ${documentLength[index]}`;
+        return `documentToValidate.length !== ${documentLength[index]} &&`;
+      });
+
+      return lengthValidatorGenerate.join(' ');
+    },
+
+    documentTypeValidators() {
```

Comment:
Where is this method being used?

Reply:
Forgot to remove it. It was used on the old validation version of the component.